src/nvim/window.c

```diff
@@ -2845,6 +2866,41 @@ int win_close(win_T *win, bool free_buf, bool force)
     return FAIL;
   }
 
+  // We need to ensure that the first split in a row is marked "left,"
+  // the last is marked "right," and the same for columns with "above"
+  // and "below"
+  if (!win->w_floating && !win->w_float_config.external) {
+    frame_T *sibling = NULL;
+    if (win->w_frame->fr_prev != NULL && win->w_frame->fr_next == NULL) {
+      sibling = win->w_frame->fr_prev;
+    } else if (win->w_frame->fr_next != NULL && win->w_frame->fr_prev == NULL) {
+      sibling = win->w_frame->fr_next;
+    }
```

Comment:
Todo: May need to restore any properties changed here if the function fails later. I should move this as far down in the function as possible without breaking things.

Reply:
Also need to handle some of this logic for neighboring windows in `win_set_config`